docs/integrations/piri-with-curio.md

```diff
@@ -0,0 +1,162 @@
+# Integrating Piri UCAN Server with Curio
+
+This guide helps existing Curio operators add Storacha network support by running a Piri UCAN server that uses Curio as the PDP backend.
+
+## Overview
+
+This integration allows you to:
+- Keep your existing Curio infrastructure
+- Add Storacha network compatibility
+- Accept uploads from Storacha clients
+- Leverage Curio's proven storage and proof capabilities
+
+Architecture:
+```
+Storacha Clients → Piri UCAN Server → Curio → Filecoin Network
+                  (This guide)      (Existing)
+```
+
+## Prerequisites
+
+Before starting, ensure you have:
+
+1. ✅ **Running Curio instance** with:
+   - HTTPS API endpoint (e.g., `https://curio.example.com`)
+   - Available storage capacity
+   - Active proof generation
+
+2. ✅ **System ready for Piri**:
+   - [Prerequisites met](../common/prerequisites.md)
+   - [Piri installed](../common/piri-installation.md)
+   - [PEM file generated](../common/pem-file-generation.md)
+
+3. ✅ **Network requirements**:
+   - Domain for UCAN server
+   - [TLS termination configured](../common/tls-termination.md)
+
+## Step 1: Configure Curio for Storacha
+
+### Add Storacha Service to Curio
+
+Curio needs to recognize the Piri UCAN server as an authorized service.
+
+1. **Extract your public key**:
+   ```bash
+   # From your service.pem file
+   openssl pkey -in service.pem -pubout
+   ```
+
+   Output example:
+   ```
+   -----BEGIN PUBLIC KEY-----
+   MCowBQYDK2VwAyEAYRpJI+aZVmKSFqYyVhMcOOeKYqBisNpReAGj4qF8kEg=
+   -----END PUBLIC KEY-----
+   ```
+
+2. **Configure Curio** to accept this key:
+   
+   Add the public key to Curio's configuration as a service named `storacha`. The exact method depends on your Curio version.
+
+3. **Restart Curio** to apply configuration
+
+## Step 2: Create Proof Set
+
+Before accepting uploads, create a proof set in Curio:
+
+```bash
+piri proofset create \
+  --key-file=service.pem \
+  --pdp-server-url=https://curio.example.com \
+  --record-keeper=0x6170dE2b09b404776197485F3dc6c968Ef948505
```

Comment:
Should this be configured as part of the piri build?

Reply:
ehh, maybe? this address is the PDP service contract address from: https://github.com/FilOzone/pdp/?tab=readme-ov-file#contracts. It's what controls the proving period, the one linked is good for testing w/ a proving period of 30min. I am going to leave this as is, but have a note to revisit in #114